cpp/leaky_core.cc

```diff
@@ -16,6 +16,25 @@
 
 namespace pyos {
 
+static List<int>* AllocSignalQueue() {
+  List<int>* ret = NewList<int>();
+  ret->reserve(kMaxSignalsInFlight);
+  return ret;
+}
+
+static List<int>* signal_queue = nullptr;
+static int last_sig_num = 0;
+static int sigwinch_num = UNTRAPPED_SIGWINCH;
+
+static void signal_handler(int sig_num) {
+  assert(signal_queue != nullptr);
+  signal_queue->append(sig_num);
```

Comment:
Did we decide if `List::append()` was signal-safe in the reserved case? If not, I can switch to a static array here and just build the list in `GetPendingSignals()`

Reply:
I think we should wrap all the global state in a class / struct

Then we can do the TSAN thing more logically.  (You could probably do it without, but for me it's easier to reason about)

I believe these are exactly the same in terms of signal safety

```
sigwinch_num = foo;

gSignalState.sigwinch_num = foo;
```

i.e. the global is at a known address, and so if the offset of the member, so the `.` doesn't actually matter for atomicity